File: automation-agent/platform-fastapi-server/app/models/api_operation_type.py
"""
API 操作类型模型
"""
from sqlalchemy import Column, String, Boolean, Text, Index
from app.db.base import Base


class ApiOperationType(Base):
    """API 操作类型表"""
    __tablename__ = "t_api_operation_type"
    
    operation_type_name = Column(String(50), nullable=False, unique=True, comment='操作类型名称')
    operation_type_desc = Column(String(255), nullable=True, comment='操作类型描述')
    operation_type_fun_name = Column(String(100), nullable=True, comment='方法名')
    operation_type_value = Column(Text, nullable=True, comment='方法体')
    is_enabled = Column(Boolean, nullable=False, default=True, index=True, comment='是否启用')
    
    __table_args__ = (
        Index('idx_operation_type_enabled', 'is_enabled'),
        {'comment': 'API操作类型表'}
    )

File: automation-agent/platform-java-server/src/main/java/com/platform/platformserver/entity/ApiOperationType.java
package com.platform.platformserver.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("t_api_operation_type")
public class ApiOperationType extends BaseEntity {
    
    @TableField("operation_type_name")
    private String operationTypeName;
    
    @TableField("operation_type_desc")
    private String operationTypeDesc;
    
    @TableField("operation_type_fun_name")
    private String operationTypeFunName;
    
    @TableField("operation_type_value")
    private String operationTypeValue;
    
    @TableField("is_enabled")
    private String isEnabled;
    
    @TableField("create_time")
    private LocalDateTime createTime;
}
